pkg/cri/streaming/portforward/httpstream.go

```diff
@@ -252,6 +258,23 @@ func (h *httpStreamHandler) portForward(p *httpStreamPair) {
 	err := h.forwarder.PortForward(h.pod, h.uid, int32(port), p.dataStream)
 	klog.V(5).Infof("(conn=%p, request=%s) done invoking forwarder.PortForward for port %s", h.conn, p.requestID, portString)
 
+	// happy path, we have successfully completed forwarding task
+	if err == nil {
+		return
+	}
+
+	// forwarding in one direction has been completed.
+	// other direction: we got RST, so we reset the data && err stream
+	if errors.Is(err, syscall.EPIPE) {
+		klog.V(5).Infof("(conn=%p, request=%s) done invoking forwarder.PortForward for port %s but got EPIPE", h.conn, p.requestID, portString)
+		p.dataStream.Reset()
+		p.errorStream.Reset()
+		streamHasBeenReset = true
+		return
+	}
+
+	// We don't know how to deal with other types of errors,
+	// try to forward them to errStream, let our user know what happened
 	if err != nil {
```

Comment:
err is definitely not nil, so it can be ignored

Reply:
fixed, thx.